packages/credential-store/src/agent/CredentialStore.ts

```diff
@@ -0,0 +1,257 @@
+  }
+
+  private readonly store: AbstractDigitalCredentialStore
+
+  constructor(options: { store: AbstractDigitalCredentialStore }) {
+    this.store = options.store
+  }
+
+  /** {@inheritDoc ICRManager.crmAddCredential} */
+  private async crsAddCredential(args: AddCredentialArgs): Promise<DigitalCredential> {
+    return await this.store.addCredential(args.credential)
+  }
+
+  /** {@inheritDoc ICRManager.updateCredentialState} */
+  private async crsUpdateCredentialState(args: UpdateCredentialStateArgs): Promise<DigitalCredential> {
+    return await this.store.updateCredentialState(args)
+  }
+
+  /** {@inheritDoc ICRManager.crmGetCredential} */
+  private async crsGetCredential(args: GetCredentialArgs): Promise<DigitalCredential> {
+    const { id } = args
+    const credential = await this.store.getCredential({ id })
+    return credential
+  }
+
+  /** {@inheritDoc ICRManager.crmGetCredentials} */
+  private async crsGetCredentials(args: GetCredentialsArgs): Promise<Array<DigitalCredential>> {
+    const { filter } = args
+    const credentials = await this.store.getCredentials({ filter })
+    return credentials.data
+  }
+
+  /** {@inheritDoc ICRManager.crmGetUniqueCredentialByIdOrHash} */
+  private async crsGetUniqueCredentialByIdOrHash(args: GetCredentialsByIdOrHashArgs): Promise<OptionalUniqueDigitalCredential> {
+    const credentials = await this.crsGetCredentials({ filter: credentialIdOrHashFilter(args.credentialRole, args.idOrHash) })
+    if (credentials.length === 0) {
+      return undefined
+    }
+    return this.toUniqueCredentials(credentials)[0]
+  }
+
+  /** {@inheritDoc ICRManager.crmGetUniqueCredentials} */
+  private async crsGetUniqueCredentials(args: GetCredentialsArgs): Promise<Array<UniqueDigitalCredential>> {
+    const credentials = await this.crsGetCredentials(args)
+    return this.toUniqueCredentials(credentials)
+  }
+
+  /** {@inheritDoc ICRManager.crmDeleteCredential} */
+  private async crsDeleteCredential(args: DeleteCredentialArgs): Promise<boolean> {
+    return this.store.removeCredential(args)
+  }
+
+  /** {@inheritDoc ICRManager.crmDeleteCredentials} */
+  private async crsDeleteCredentials(args: DeleteCredentialsArgs): Promise<number> {
+    const credentials = await this.crsGetCredentials(args)
+    let count = 0
+    for (const credential of credentials) {
+      const result = await this.store.removeCredential({ id: credential.id })
+      if (result) {
+        count++
+      }
+    }
+    return count
+  }
+
+  /**
+   * Returns a list of UniqueDigitalCredentials that match the given filter based on the claims they contain.
+   * @param args
+   */
+  private async crsGetCredentialsByClaims(args: GetCredentialsByClaimsArgs): Promise<Array<UniqueDigitalCredential>> {
+    const digitalCredentials = await this.crsGetUniqueCredentials({
+      filter: [
+        {
+          documentType: DocumentType.VC, // TODO does crmGetCredentialsByClaims need to support VPs as well?
```

Comment:
At least C should be supported, but a ticket should be created to indeed also support (V)P

Reply:
fixed